<launch>
  <arg name="mn" default=""/>
  <arg name="mc_x" default="0"/>
  <arg name="mc_y" default="0"/>
  <arg name="mb_w" default="0"/>
  <arg name="mb_h" default="0"/>
  <arg name="ts"/>
  <arg name="sp_x"/>
  <arg name="sp_y"/>
  <arg name="pn"/>
  <arg name="lv"/>
  <arg name="pav"/>
  <arg name="nav"/>
  <arg name="pr_c"/>
  <arg name="pr_s"/>
  <arg name="af_r"/>
  <arg name="e_rd"/>
  <arg name="e_md"/>
  <arg name="e_p"/>

  <node pkg="wandrian" type="run" name="run" output="screen">
    <remap from="run/power" to="mobile_base/commands/motor_power"/>
    <remap from="run/velocity" to="mobile_base/commands/velocity"/>
    <remap from="run/odometry" to="odom"/>
    <remap from="run/laser" to="scan"/>

    <param name="map_name" value="$(arg mn)" type="string"/>
    <param name="map_center_x" value="$(arg mc_x)" type="double"/>
    <param name="map_center_y" value="$(arg mc_y)" type="double"/>
    <param name="map_boundary_width" value="$(arg mb_w)" type="double"/>
    <param name="map_boundary_height" value="$(arg mb_h)" type="double"/>
    <param name="tool_size" value="$(arg ts)" type="double"/>
    <param name="starting_point_x" value="$(arg sp_x)" type="double"/>
    <param name="starting_point_y" value="$(arg sp_y)" type="double"/>
    <param name="plan_name" value="$(arg pn)" type="string"/>
    <param name="linear_velocity" value="$(arg lv)" type="double"/>
    <param name="positive_angular_velocity" value="$(arg pav)" type="double"/>
    <param name="negative_angular_velocity" value="$(arg nav)" type="double"/>
    <param name="proportion_ranges_count" value="$(arg pr_c)" type="double"/>
    <param name="proportion_ranges_sum" value="$(arg pr_s)" type="double"/>
    <param name="augmentation_factor_range" value="$(arg af_r)" type="double"/>
    <param name="epsilon_rotational_direction" value="$(arg e_rd)" type="double"/>
    <param name="epsilon_motional_direction" value="$(arg e_md)" type="double"/>
    <param name="epsilon_position" value="$(arg e_p)" type="double"/>

    <param name="linear_velocity_step" value="0.15" type="double"/>
    <param name="linear_velocity_max" value="1.5" type="double"/>
    <param name="angular_velocity_step" value="0.75" type="double"/>
    <param name="angular_velocity_max" value="3" type="double"/>
  </node>
</launch>
